### PR TITLE
try to fix codecov upload -> TRAVIS authorization problem

### PR DESCRIPTION
**Pull request by sappelhoff**
as discussed in #140 

not sure if it works, but worth a try.

### CODE REVIEW

**Comment by sappelhoff** [comment]
@skjerns the problem is that Travis is not building at all: https://travis-ci.com/github/holgern/pyedflib/requests

Some authorization problem:

image

**Comment by sappelhoff** [comment]
relevant: https://stackoverflow.com/a/41327657/5201771

File: .travis.yml [hunk 1]
```diff
@@ -177,16 +177,12 @@ script:
...
-        pytest --pyargs pyedflib --cov=pyedflib --cov-config=../.coveragerc
-        cp .coverage ..
-        popd
+        python -m pytest --pyargs pyedflib --cov=pyedflib --cov-config=.coveragerc --cov-report=xml --verbose
     fi
 
 after_success:
   - |
     if [ "${MANYLINUX}" == "0" ]; then
-        codecov
```
Comment (skjerns) [inline .travis.yml:189]: sure this is correct?

Comment (sappelhoff) [reply .travis.yml:189]: yes, `codecov` only gathers the coverage report and submits it to codecov:

> Find coverage reports for all the languages below, gather them and submit them to Codecov.

source: https://github.com/codecov/codecov-python

The same is achieved by the small bash snippet a few lines below: 

> bash <(curl -s https://codecov.io/bash)

Last time I checked, that bash snippet was the recommended mode of uploading reports to codecov.

Removing the python `codecov` package here just prevents duplication.

**Comment by skjerns** [comment]
@holgern can you check the authorization on travis/coverage?

however, it's a bit strange as the regular builds work fine

**Comment by sappelhoff** [comment]
> however, it's a bit strange as the regular builds work fine

but the last Travis CI builds are more than a month past - despite several merges (other than PRs) in between :thinking: 

Or are both of us looking at different data? --> https://travis-ci.com/github/holgern/pyedflib/builds

File: .travis.yml (hunk 1 above)
Comment (skjerns) [reply .travis.yml:189]: thanks!

**Comment by holgern** [comment]
@skjerns I checked the authorization and they look fine. Should we change the coverage tool?, maybe there is a better one or the coverage should not be uploaded from travis.

**Comment by skjerns** [comment]
yes, that seems like a good idea. currently there are three CI tools active anyway (travis, circle, AppVeyor ?), I guess one or two should be sufficient.

I guess the coverage should be able to upload from either of those

**Comment by skjerns** [comment]
Alternatively we could just use github actions for running tests?

+ best integration and no authorization problems
- less verbose interface and might be a bit more cumbersome if stuff fails

example from one of my other repos 
https://github.com/Unisens/pyunisens/blob/master/.github/workflows/Tests.yml

**Comment by skjerns** [comment]
I created a workflow for codecov and it works now. 

However I'm not sure if the Cython code was validated before, and if that's now missing?

I also removed all traces of travis

**Comment by sappelhoff** [comment]
nice, thanks for fixing this @skjerns

**Pull request closed without merging**